web/app/src/components/AnnouncementBanner.vue

```diff
@@ -232,6 +237,55 @@ const getTypeClasses = (type) => {
   return typeConfigs[type] || typeConfigs.none
 }
 
+const escapeHtml = (value) => {
+  if (value === null || value === undefined) {
+    return ''
+  }
+
+  return String(value)
+    .replace(/&/g, '&amp;')
+    .replace(/</g, '&lt;')
+    .replace(/>/g, '&gt;')
+    .replace(/"/g, '&quot;')
+    .replace(/'/g, '&#39;')
+}
```

Comment:
There's probably a more secure way to do this and I'm sure some security scan will complain, but since the content of the announcement is controlled by the configuration file, XSS isn't a concern.